src/phpDocumentor/Phpdoc2XmlToArrayConverter.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace phpDocumentor;
+
+final class Phpdoc2XmlToArrayConverter implements XmlConverter
+{
+    /**
+     * @var \SimpleXMLElement
+     */
+    private $xml;
+
+    /**
+     * @param \SimpleXMLElement $xml
+     */
+    public function __construct(\SimpleXMLElement $xml)
+    {
+        $this->xml = $xml;
+    }
+
+    /**
+     * Converts the phpDocumentor2 configuration xml to an array.
+     *
+     * @return array
+     */
+    public function convert()
+    {
+        $phpDocumentor = $this->xml;
+
+        $extensions         = [];
+        $markers            = [];
+        $visibility         = 'public';
+        $defaultPackageName = 'Default';
+        $template           = 'clean';
+        $ignoreHidden       = true;
+        $ignoreSymlinks     = true;
+
+        if (isset($phpDocumentor->parser)) {
+            $extensions = $this->buildExtensionsPart($phpDocumentor->parser);
+            $markers    = $this->buildMarkersPart($phpDocumentor->parser);
+
+            $visibility         = ((string) $phpDocumentor->parser->visibility) ?: $visibility;
+            $defaultPackageName = ((string) $phpDocumentor->parser->{'default-package-name'}) ?: $defaultPackageName;
+            $template           = ((string) $phpDocumentor->transformations->template->attributes()->name) ?: $template;
+
+            if (isset($phpDocumentor->parser->files)) {
+                if (isset($phpDocumentor->parser->files->{'ignore-hidden'})) {
+                    $ignoreHidden = filter_var($phpDocumentor->parser->files->{'ignore-hidden'}, FILTER_VALIDATE_BOOLEAN);
+                }
+
+                if (isset($phpDocumentor->parser->files->{'ignore-symlinks'})) {
+                    $ignoreSymlinks = filter_var($phpDocumentor->parser->files->{'ignore-symlinks'}, FILTER_VALIDATE_BOOLEAN);
+                }
+            }
+        }
+
+        $outputDirectory = ((string) $phpDocumentor->parser->target) ?: 'file://build/docs';
+        $sourcePath      = ((string) $phpDocumentor->parser->files->directory) ?: 'src';
```

Comment:
Missing the `file` element support in the `$phpDocumentor->parser->files` element